openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/FileLayerProvider.scala

```diff
@@ -1415,13 +1416,27 @@ class FileLayerProvider private(openSearch: OpenSearchClient, openSearchCollecti
        * Several edge cases to cover:
        *  - if feature extent is whole world, it may be invalid in target crs
        *  - if feature is in utm, target extent may be invalid in feature crs
-       *  this is why we take intersection
+       *  this is why we take intersection.
+       *  We convert both extents to a common CRS before taking the intersection.
+       *  If one of the CRSes can cover the whole world (non-UTM), this will be used as common CRS.
+       *  We give priority to use the target CRS as common one, because the intersection will be converted to it anyway
        */
-      val targetExtentInLatLon = targetExtent.reproject(feature.crs.get)
-      val featureExtentInLatLon = feature.rasterExtent.get.reproject(feature.crs.get,LatLng)
-
-      val intersection = featureExtentInLatLon.intersection(targetExtentInLatLon).map(_.buffer(1.0)).getOrElse(featureExtentInLatLon)
-      val tmp = expandToCellSize(intersection.reproject(LatLng, targetExtent.crs), theResolution)
+      val featureIsUTM = feature.crs.get.proj4jCrs.getProjection.getName == "utm"
+      val targetIsUTM = targetExtent.crs.proj4jCrs.getProjection.getName == "utm"
+      val commonCrs = if (!targetIsUTM) targetExtent.crs
+      else if (!featureIsUTM) feature.crs.get
+      else targetExtent.crs // Avoid conversion imprecision by intersecting directly in the target CRS
+
+      val featureExtentInCommonCRS = feature.rasterExtent.get.reproject(feature.crs.get, commonCrs)
+      val targetExtentInCommonCRS = targetExtent.extent.reproject(targetExtent.crs, commonCrs)
+
+      val intersection = featureExtentInCommonCRS.intersection(targetExtentInCommonCRS).map(_.buffer(1.0))
+      val intersectionTargetCrs = intersection match {
+        case None => targetExtent.extent.reproject(targetExtent.crs, targetExtent.crs)
```

Comment:
I'm not sure if we should keep this case, can you open a new issue to look into simply discarding the feature when there is no intersection?
Adding a log line could help to reveal if this actually occurs in production, or just an edge case in the code.

Reply:
Just checked, it gets in this case when an extent gets wrapped badly around the antimeridian after reproject. 
We would be more in control if we double check the project result and wrap it manually if needed
![image](https://github.com/user-attachments/assets/3b1128be-07e4-4d99-8983-7386f00a4788)

Reply:
Maybe only changing this could be the fix.
The UTM projected to latlon becomes extent is `[-179, 177]`, which is wrong.
It should be represented as:
`[177, 181]` or maybe even `[-183, -179]`
But I guess some code prefers to keep everything within `[-180, 180]`, this would need some extra checking.

Reply:
The latlon warp is avoided by using `safeReproject()`

Reply:
Also the feature will now get discarded when not intersecting.